### ISSUE TITLE
Unable to use SwiftLint as Swift Package Manager dependency

### ISSUE THREAD

**Issue by rahulmcatct** [realm/SwiftLint#4746]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

I am unable to add latest SwiftLint as SPM dependency.
Getting the following error:
<img width="1507" alt="MicrosoftTeams-image (2)" src="https://user-images.githubusercontent.com/1086637/216896536-8789c3dd-0e53-46d9-b54b-52ccfda7a759.png">

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
* Paste your configuration file:

```yml
# insert yaml contents here
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.
* Which Xcode version are you using (check `xcodebuild -version`)?
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
// This triggers a violation:
let foo = try! bar()
```

**Comment by benfrearson** [comment]
Just commenting to say I've got the same issue. Reverting to a pre 0.50.0 release works for me

**Comment by shibotong** [comment]
Maybe you can change version to `0.50.3` manually. `0.9.2` is installed which should be long time ago. Don't use default latest version configuration in SPM. It will treat `0.50` < `0.9`.

**Comment by SimplyDanny** [comment]
@shibotong is right. Version 0.9.2 is nearly 7 years old! Please try with the most recent version 0.50.3.

**Comment by mt00chikin** [comment]
I'm experiencing the same package resolution error and was able to get it to resolve by using the `.package(url:revision:)` API to point directly to the 0.50.3 release commit (a876e860ee0e166a05428f430888de5d798c0f8d). I used [this example](https://forums.swift.org/t/dependencies-could-not-be-resolved-because-package-yeelampa-is-required-using-a-stable-version-but-yeelampa-depends-on-an-unstable-version-package-requester-and-root-depends-on-yeelampa-0-1-0/53840) in the Swift forums to test this out.

Is it possible that the tag is somehow messed up? If I try to specify the exact version of 0.50.3 in SPM, I get this error in Xcode: 

> Failed to resolve dependencies Dependencies could not be resolved because package 'swiftlint' is required using a stable-version but 'swiftlint' depends on an unstable-version package 'swift-syntax' and 'buildtools' depends on 'swiftlint' 0.50.3.

**Comment by SimplyDanny** [comment]
SwiftLint switched to a somehow stable release tag of SwiftSyntax only after the 0.50.3 release. So the "unstable version" should disappear with the next SwiftLint release.

**Comment by mdaymard-gpsw** [comment]
What can we do in the meantime to solve this issue ? I get the same error, and I've tried with `0.49.1` and `0.48.0`, and after deleting the derived data

**Comment by jpsim** [comment]
Could everyone please try the latest release candidate to see if this improves things? https://github.com/realm/SwiftLint/releases/tag/0.51.0-rc.2

**Comment by vladvlasov256** [comment]
@jpsim "Failed to resolve dependencies" error has gone for me, I specified the exact version "0.51.0-rc.2"

**Comment by mt00chikin** [comment]
I also just tested it by pointing to `0.51.0-rc.2` and SPM resolved successfully

**Comment by jpsim** [comment]
Going to mark this as closed, thanks for checking @vladvlasov256 and @mt00chikin! 🙏
